internal/version/version.go

```diff
@@ -53,9 +53,6 @@ type BuildInfo struct {
 
 // GetVersion returns the semver string of the version
 func GetVersion() string {
-	if metadata == "" {
-		return version
-	}
```

Comment:
When we have a stable release of Helm the `metadata` is an empty string. Without this case we would have versions returned such as `v3.5.3+` instead of `v3.5.3`.

Can you provide examples and cases to what you're intending?